openmc/mgxs/mgxs.py

```diff
@@ -1884,13 +1886,19 @@ def tallies(self):
             energy = openmc.Filter('energy', group_edges)
             energyout = openmc.Filter('energyout', group_edges)
 
-            # Create a list of scores for each Tally to be created
-            if self.correction == 'P0':
-                scores = ['flux', 'scatter', 'scatter-P1']
-                filters = [[energy], [energy, energyout], [energyout]]
-            else:
-                scores = ['flux', 'scatter']
-                filters = [[energy], [energy, energyout]]
+            # Create lists of scores, filters for each Tally to be created
+            scores = ['flux']
+            filters = [[energy]]
+
+            # Create separate tallies for each moment
+            for moment in range(self.legendre_order+1):
+                scores.append('scatter-{}'.format(moment))
```

Comment:
It is true that "scatter-PN" is more efficient than "scatter-X".  The cycling through the orders is an inner loop for "scatter-PN", while if you had "scatter-0 scatter-1 scatter-2 ..." then you have to cycle through the score loop each time and incur that overhead.